src/printer.c

```diff
@@ -143,47 +143,25 @@ static void print_proto_ipv6_ext_mobility(void *data, size_t size, FILE *o)
 
 static void print_l4(uint8_t type, void *data, size_t size, FILE *o)
 {
-	switch (type) {
-	case IPPROTO_TCP:
-		print_proto_tcp(data, size, o);
-		break;
-	case IPPROTO_UDP:
-		print_proto_udp(data, size, o);
-		break;
-	case IPPROTO_ICMP:
-		print_proto_icmp(data, size, o);
-		break;
-	case IPPROTO_ICMPV6:
-		print_proto_icmpv6(data, size, o);
-		break;
-	case IPPROTO_IGMP:
-		print_proto_igmp(data, size, o);
-		break;
-	case 0x00:
-		print_proto_ipv6_ext_hbh(data, size, o);
-		break;
-	case 0x2b:
-		print_proto_ipv6_ext_routing(data, size, o);
-		break;
-	case 0x2c:
-		print_proto_ipv6_ext_fragment(data, size, o);
-		break;
-	case 0x32:
-		print_proto_ipv6_ext_esp(data, size, o);
-		break;
-	case 0x33:
-		print_proto_ipv6_ext_ah(data, size, o);
-		break;
-	case 0x3c:
-		print_proto_ipv6_ext_destination(data, size, o);
-		break;
-	case 0x87:
-		print_proto_ipv6_ext_mobility(data, size, o);
-		break;
-	default:
+	static void (*ptr_print_proto[136])(void *data, size_t size, FILE *);
```

Comment:
you should declare this variable as local storage.
The initialization bellow should be done only once, here you set ptr_print_proto at each print_l4 call.
ptr_print_proto is not a really good name, we can see that it's a pointer, so it's useless to add ptr in your variable name.